app/controllers/users_controller.rb

```diff
@@ -1,32 +1,55 @@
 class UsersController < ApplicationController
+
+  rescue_from ActiveRecord::RecordNotFound, with: :category_not_found
+  
+  #getting all users
   def index
-    user = User.all
-    render json: user, status: :ok
+    users = User.all
+    render json: users, status: :ok
   end
 
+  #posting a new user
   def create
-    user = User.create!(user_params)
+    user = User.create(user_params)
     render json: user, status: :created
   end
 
+  #patching a new user
   def update
-    user = User.update(user_params)
-    render json: user, status: :ok
+    user = find_user
+    user.update(user_params)
+    render json: user, status: :accepted
   end
 
+  #getting one user
   def show
-    user = User.find_by(id: params[:id])
-    if user
-      render json: user, status: :ok
-    else
-      render json: { error: "User not found" }, status: :not_found
-    end
+    user = find_user
+    render json: user, status: :ok
   end
 
+  #deleting a user
   def destroy
-    user = User.find_by(id: params[:id])
+    user = find_user
     user.destroy
     head :no_content
   end
 
+  #the private methods
+  private
+
+  #find user
+  def find_user
+    User.find(params(:id))
+  end
+
+  #allowed params
+  def user_params
```

Comment:
When creating a user, we take the params, first_name, last_name, username, email, phone_number, role, password, password_confirmation. The parameters permitted are misinterpreted and others are missing
We need to have something like `params.permit(:first_name, :last_name, :username, :email, :phone_number, :role, :password, :password_confirmation)`